WordPress/src/main/java/org/wordpress/android/reader/savedposts/provider/ReaderSavedPostsProvider.kt

```diff
@@ -0,0 +1,56 @@
+package org.wordpress.android.reader.savedposts.provider
+
+import android.database.Cursor
+import android.net.Uri
+import org.wordpress.android.WordPress
+import org.wordpress.android.datasets.wrappers.ReaderPostTableWrapper
+import org.wordpress.android.models.ReaderTag
+import org.wordpress.android.models.ReaderTagType.BOOKMARKED
+import org.wordpress.android.provider.query.QueryContentProvider
+import org.wordpress.android.provider.query.QueryResult
+import org.wordpress.android.util.publicdata.ClientVerification
+import org.wordpress.android.util.publicdata.JetpackPublicData
+import org.wordpress.android.util.signature.SignatureNotFoundException
+import javax.inject.Inject
+
+class ReaderSavedPostsProvider : QueryContentProvider() {
+    @Inject lateinit var queryResult: QueryResult
+    @Inject lateinit var readerPostTableWrapper: ReaderPostTableWrapper
+    @Inject lateinit var jetpackPublicData: JetpackPublicData
+    @Inject lateinit var clientVerification: ClientVerification
+
+    override fun onCreate(): Boolean {
+        return true
+    }
+
+    @Suppress("SwallowedException")
+    override fun query(
+        uri: Uri,
+        projection: Array<out String>?,
+        selection: String?,
+        selectionArgs: Array<out String>?,
+        sortOrder: String?
+    ): Cursor? {
+        inject()
+        return context?.let {
```

Comment:
`JetpackProviderSyncFeatureConfig` was already merged to `trunk`. Would it make sense to add it now or we're doing it in a different PR?